infra/Pos/Network/Types.hs

```diff
@@ -47,21 +48,45 @@ import           GHC.Show                              (Show(..))
 
 -- | Information about the network in which a node participates.
 data NetworkConfig kademlia = NetworkConfig
-    { ncTopology    :: !(Topology kademlia)
+    { ncTopology      :: !(Topology kademlia)
       -- ^ Network topology from the point of view of the current node
-    , ncDefaultPort :: !Word16
+    , ncDefaultPort   :: !Word16
       -- ^ Port number to use when translating IP addresses to NodeIds
-    , ncSelfName    :: !(Maybe NodeName)
+    , ncSelfName      :: !(Maybe NodeName)
       -- ^ Our node name (if known)
+    , ncEnqueuePolicy :: !(OQ.EnqueuePolicy NodeId)
+    , ncDequeuePolicy :: !OQ.DequeuePolicy
+    , ncFailurePolicy :: !(OQ.FailurePolicy NodeId)
     }
-  deriving (Show)
 
-defaultNetworkConfig :: Topology topology -> NetworkConfig topology
+instance Show kademlia => Show (NetworkConfig kademlia) where
+    show = show . showableNetworkConfig
+
+data ShowableNetworkConfig kademlia = ShowableNetworkConfig {
+      sncTopology    :: !(Topology kademlia)
+    , sncDefaultPort :: !Word16
+    , sncSelfName    :: !(Maybe NodeName)
+    }
+    deriving (Show)
+
+showableNetworkConfig :: NetworkConfig kademlia -> ShowableNetworkConfig kademlia
+showableNetworkConfig NetworkConfig {..} =
+    let sncTopology    = ncTopology
+        sncDefaultPort = ncDefaultPort
+        sncSelfName    = ncSelfName
+    in  ShowableNetworkConfig {..}
+
+defaultNetworkConfig :: Topology kademlia -> NetworkConfig kademlia
 defaultNetworkConfig ncTopology = NetworkConfig {
-      ncDefaultPort = 3000
-    , ncSelfName    = Nothing
+      ncDefaultPort   = 3000
+    , ncSelfName      = Nothing
+    , ncEnqueuePolicy = Policy.defaultEnqueuePolicy ourNodeType
```

Comment:
I guess here is where the `topologyEnqueuePolicy` and co from #1286 could be used?